contracts/cw4-stake/src/state.rs

```diff
@@ -6,13 +6,15 @@ use cw0::Duration;
 use cw4::TOTAL_KEY;
 use cw_controllers::{Admin, Claims, Hooks};
 use cw_storage_plus::{Item, Map, SnapshotMap, Strategy};
+use cw20::{Balance, Cw20Coin};
+use cosmwasm_std::Coin;
 
 pub const CLAIMS: Claims = Claims::new("claims");
 
 #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
 pub struct Config {
     /// denom of the token to stake
-    pub denom: String,
+    pub denom: Balance,
```

Comment:
Why a balance here? This is a different object, with no amount:

```rust
pub enum Denom {
  Native(String),
  Cw20(HumanAddress),
}
```

This will just have one object of one type and be much simpler to compare with.
I think with proper data structures, the logic in the contract can become much simpler

Reply:
Do you want me to declare this new enum in CW4 package?